frontend/lib/identification/identification_qr_content_parser.dart

```diff
@@ -0,0 +1,127 @@
+      } on Exception catch (e) {
+        return QRCodeParseResult(
+            hasError: true,
+            internalErrorMessage:
+                "Failed to decode base64 string from qr code, "
+                "probably not base64 encoded. Message: ${e.toString()}",
+            userErrorMessage:
+                "Der Inhalt des eingescannten Codes kann nicht verstanden "
+                "werden. Vermutlich handelt es sich um einen QR Code, "
+                "der nicht für die Ehrenamtskarte App generiert wurde.");
+      }
+      CardActivateModel cardActivateModel;
+      try {
+        cardActivateModel = CardActivateModel.fromBuffer(rawProtobufData);
+      } on Exception catch (e) {
+        return QRCodeParseResult(
+            hasError: true,
+            internalErrorMessage:
+                "Failed to parse CardActivateModel from base64 encoded data. "
+                "Message: ${e.toString()}",
+            userErrorMessage:
+                "Der Inhalt des eingescannten Codes kann nicht verstanden "
+                "werden. Vermutlich handelt es sich um einen QR Code, "
+                "der nicht für die Ehrenamtskarte App generiert wurde.");
+      }
+
+      final fullName = cardActivateModel.fullName;
+      if (fullName == null) {
+        return createFieldMissingError("fullName", "fullNameMissing");
+      }
+      final hashSecret = cardActivateModel.hashSecret;
+      if (hashSecret == null) {
+        return createFieldMissingError("randomBytes", "randomBytesMissing");
+      }
+      final unixInt64ExpirationDate = cardActivateModel.expirationDate;
+      int unixExpirationDate;
+      if (unixInt64ExpirationDate != null) {
+        try {
+          unixExpirationDate = unixInt64ExpirationDate.toInt();
+          if (unixExpirationDate == null) {
+            throw Exception(["unixExpirationDate was null"]);
+          }
+        } on Exception catch (e) {
+          return QRCodeParseResult(
+              hasError: true,
+              internalErrorMessage: "Failed to parse unixExpirationDate, "
+                  "Message: ${e.toString()}",
+              userErrorMessage: "Beim Verarbeiten des Ablaufdatums ist ein "
+                  "unerwarteter Fehler aufgetreten.");
+        }
+      }
+      if (cardActivateModel.cardType == CardActivateModel_CardType.STANDARD &&
+          unixExpirationDate == null) {
+        return QRCodeParseResult(
+            hasError: true,
+            internalErrorMessage: "Got card type "
+                "${cardActivateModel.cardType.toString()} but "
+                "unixExpirationDate was not set",
+            userErrorMessage: "Die eingescannte Karte enthält kein Ablauf-"
+                "datum, obwohl dies für die blaue Ehrenamtskarte erforderlich"
+                " ist. Vermutlich ist beim Erstellen der "
+                "digitalen Ehrenamtskarte ein Fehler passiert.");
+      }
+      final cardType = cardActivateModel.cardType.toString();
```

Comment:
Should this be toString()? Is this only for display or also for generating a hash?

Reply:
I will look at this in #222